scripts/notebook-normalizer/qiskit_docs_notebook_normalizer/__init__.py

```diff
@@ -125,14 +126,16 @@ def normalize_notebook(
                 # keep looking at other cells
                 return NormalizationNeeded(nb=nb, images=[])
 
-    if change_made:
+    has_unused_images = any(image for image in get_unused_images(image_folder, nb))
```

Comment:
You can speed up this algorithm by using a set difference. Right now, it's O(n^2) because you have a nested for loop: for every image, loop through every output cell. Instead, pre-compute:

- all images in the folder
- all used image names in the notebook

Then, take the set difference of `all_image_files - used_images`. O(n)

Reply:
That's cool, I wasn't aware of that approach.

Implementing it here does have a trade off: We'd need to use a regex (or similar) to extract the filenames from the outputs, which means we need to make assumptions about the form of the mdx component (`<Image src="...`). This should be fine for now, but it could cause problems if we ever change the MDX component (e.g. changing arrangement of properties).

Do you think the better scaling is worth it?

Reply:
Ah, touché. I didn't think about that being more brittle with extracting the image names. Your approach sounds good!